### PR TITLE
fix crash when displaying an error message

### PR DESCRIPTION
**Pull request by adrpo**
to test:
- git clone https://github.com/OpenModelica/BUAVSystem.git
- build model:
  BUAVSystem.Examples.ValidationCases.IntegratedSystemValidationMar6
- crashes in:
  RemoveSimpleEquations.mo:3156 because src is defined only one one branch

### CODE REVIEW

File: OMCompiler/Compiler/BackEnd/RemoveSimpleEquations.mo [hunk 1]
```diff
@@ -3137,12 +3137,12 @@ algorithm
     then BackendVariable.setVarStartValueOption(v, start);
 
     case (_, true, (_, values), BackendDAE.SHARED(globalKnownVars=globalKnownVars)) algorithm
-      v := BackendVariable.setVarFixed(inVar, true);
```
Comment (kabdelhak) [inline OMCompiler/Compiler/BackEnd/RemoveSimpleEquations.mo:3140]: That needs to happen independent of the status of the dumping flag, why did you move it inside?

File: OMCompiler/Compiler/BackEnd/RemoveSimpleEquations.mo [hunk 2]
```diff
@@ -3152,9 +3152,10 @@ algorithm
           end if;
           str := str + " * Candidate: " + ComponentReference.printComponentRefStr(cr) + "(start = " + ExpressionDump.printExpStr(startExp) + ", confidence number = " + intString(i) + ")\n";
         end for;
+        Error.addCompilerError(str);
       end if;
-      Error.addCompilerError(str);
```
Comment (kabdelhak) [inline OMCompiler/Compiler/BackEnd/RemoveSimpleEquations.mo:3156]: This one seems to be an oversight on my part, thanks for the fix!

File: OMCompiler/Compiler/BackEnd/RemoveSimpleEquations.mo (hunk 1 above)
Comment (adrpo) [reply OMCompiler/Compiler/BackEnd/RemoveSimpleEquations.mo:3140]: ah, i didn't get that. i'll move it back-

**Comment by adrpo** [comment]
I'll just merge it as it is, then you can hack it as you like.

**Comment by adrpo** [comment]
i also gave you access to my repo so you can push directly to my PRs.
